src/Microsoft.AspNet.OData.Shared/Query/DefaultSkipTokenHandler.cs

```diff
@@ -0,0 +1,420 @@
+        /// <param name="context">Serializer context</param>
+        /// <returns>Returns the URI for NextPageLink. If a null object is passed for the instance, resorts to the default paging mechanism of using $skip and $top.</returns>
+        public override Uri GenerateNextPageLink(Uri baseUri, int pageSize, Object instance, ODataSerializerContext context)
+        {
+            if (context == null)
+            {
+                return null;
+            }
+
+            if (pageSize <= 0)
+            {
+                return null;
+            }
+
+            Func<object, string> skipTokenGenerator = null;
+            IList<OrderByNode> orderByNodes = null;
+            ExpandedReferenceSelectItem expandedItem = context.CurrentExpandedSelectItem;
+            IEdmModel model = context.Model;
+
+            DefaultQuerySettings settings = context.QueryContext.DefaultQuerySettings;
+            if (settings.EnableSkipToken)
+            {
+                if (expandedItem != null)
+                {
+                    // Handle Delta resource; currently not value based.
+                    if (TypedDelta.IsDeltaOfT(context.ExpandedResource.GetType()))
+                    {
+                        return GetNextPageHelper.GetNextPageLink(baseUri, pageSize);
+                    }
+
+                    if (expandedItem.OrderByOption != null)
+                    {
+                        orderByNodes = OrderByNode.CreateCollection(expandedItem.OrderByOption);
+                    }
+
+                    skipTokenGenerator = (obj) =>
+                    {
+                        return GenerateSkipTokenValue(obj, model, orderByNodes);
+                    };
+
+                    return GetNextPageHelper.GetNextPageLink(baseUri, pageSize, instance, skipTokenGenerator);
+                }
+
+                if (context.QueryOptions != null && context.QueryOptions.OrderBy != null)
+                {
+                    orderByNodes = context.QueryOptions.OrderBy.OrderByNodes;
+                }
+
+                skipTokenGenerator = (obj) =>
+                {
+                    return GenerateSkipTokenValue(obj, model, orderByNodes);
+                };
+            }
+
+            return context.InternalRequest.GetNextPageLink(pageSize, instance, skipTokenGenerator);
+        }
+
+        /// <summary>
+        /// Generates a string to be used as the skip token value within the next link.
+        /// </summary>
+        /// <param name="lastMember"> Object based on which SkipToken value will be generated.</param>
+        /// <param name="model">The edm model.</param>
+        /// <param name="orderByNodes">List of orderByNodes used to generate the skiptoken value.</param>
+        /// <returns>Value for the skiptoken to be used in the next link.</returns>
+        protected virtual string GenerateSkipTokenValue(Object lastMember, IEdmModel model, IList<OrderByNode> orderByNodes)
```

Comment:
>protected [](start = 8, length = 9)

Make this private for now and have an issue to track exposing this method along with other methods to allow for customization serialization and deserialization of the name value pairs.